### PR TITLE
Add region in IBM COS config

### PR DESCRIPTION
**Pull request by JosepSampe**
- Add region parameter in IBM COS config.

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by gilv** [comment]
@JosepSampe let me check few things...hold on to merge it please

**Comment by gilv** [comment]
@JosepSampe there are cross region and per region endpoints..how it will work from now on?

**Comment by gilv** [comment]
@JosepSampe there are also might be private cloud endpoints, for those who install COS on prem

**Comment by JosepSampe** [comment]
>>@JosepSampe there are cross region and per region endpoints..how it will work from now on?

You only need to write the region of your bucket, either regional or cross regional, for example:
```yaml
ibm_cos:
   region: us-east  # region
```
or
```yaml
ibm_cos:
   region: us  #cross-region
```

>>@JosepSampe there are also might be private cloud endpoints, for those who install COS on prem

Can IBM COS be installed on-prem? never noticed bout this. Where are the instructions?

**Comment by gilv** [comment]
@JosepSampe it's not enough...bucket may be cross region or bucket may be regional. Those are different endpoints, not only "region" name. COS on Prem - you buy it :)

**Comment by JosepSampe** [comment]
I don't understand why its not enough.. If you have a cross region bucket you write for example **us**, if you have single region bucket you write for exemple **us-east.**, as simple as this. What else is required?

**Comment by gilv** [comment]
@JosepSampe you are right, it will work. But it will not work if we need to provide some custom endpoint to the COS on prem

**Comment by JosepSampe** [comment]
@gilv We can keep endpoint in confg. If a user writes endpoints, lithops will use them, if no, it will use region and create endpoints automatically. What do you think?

**Comment by gilv** [comment]
@JosepSampe just don't make them deprecated. Leave private and public as is. And we can use your simplified approach for just write region name and that's it. Are you fine with it? (One of the https://www.ibm.com/products/cloud-object-storage products is on-prem)

**Comment by JosepSampe** [comment]
@gilv Now I reverted previous config, so you can configure it by using **endpoint** and **preivate_endpoint** (as usual) or by using the region

**Comment by JosepSampe** [comment]
@gilv Are you fine with this now? :)

File: config/storage/ibm_cos.md [hunk 1]
```diff
@@ -34,8 +34,7 @@ Lithops with IBM COS as storage backend.
         storage_backend: ibm_cos
        
     ibm_cos:
-       endpoint   : <REGION_ENDPOINT>  
-       private_endpoint : <PRIVATE_REGION_ENDPOINT>
```
Comment (gilv) [inline config/storage/ibm_cos.md:38]: @JosepSampe add them back. May be put in comment the usage, etc ?

Comment (JosepSampe) [reply config/storage/ibm_cos.md:38]: @gilv Like in the rest of compute and storage backends, this config example intend to show minimum configuration needed. All available keys are specified in the table below. If we add the keys here, we can delete the table, I mean, there is no need to duplicate things.

**Comment by JosepSampe** [comment]
@gilv I'm still confused with this on-prem installation you mention. I agree IBM COS can be installed in a private cluster, But current IBM COS instruccions make reference to IBM Cloud dashboard, so I guess if someone could configure an on-porem COS following these instructions. 

I think if you want to support on-prem IBM COS installation you should add a specific section in instructions, because from my part there are alot of things that I don't know how works:
- Have endpoints in a pon-prem installation similar format? 
- On-prem installation has regions? 
- What type of auth an on-prem installation is using? api_key? access and secret key?
- On-prem installations are only private? or are they also public?
- On-prem installation have private, public and direct endpoints? or is the cluster accessible with a unique endpoint?

**Comment by gilv** [comment]
@JosepSampe it's not that different...customer just buy COS and that's it. No need private or public on prem - both points to the same COS address. It uses access and secret key. It all already works..just this PR when removed endpoints broke it. But now all is good. I don't have more comments, i also agree with your comment on documentation on keys. I am merging it

**Pull request merged**